### PR TITLE
feat(GEN-95|HELIX-1134): add datadog and mixpanel support 

### PR DESCRIPTION
**Pull request by danidomi**
adds support for datadog (Profiles and Spans)
adds [mixpanel](https://docs.mixpanel.com/docs/what-is-mixpanel) support (BAU metrics)

this PR also merges the indexer client of metrics in here!

https://github.com/InjectiveLabs/injective-indexer/blob/dev/metrics/client.go

## Summary by CodeRabbit

- **New Features**
	- Introduced MixPanel integration for enhanced analytics.
	- Added profiling capabilities with DataDog for better performance monitoring.
	- Enhanced error reporting and event tracking with new functions in the metrics framework.
	- New metrics reporting functions for counting, timing, and gauging metrics.

- **Bug Fixes**
	- Improved error handling in function calls.

- **Chores**
	- Updated dependencies for better observability and error tracking capabilities.

### CODE REVIEW

**Review by proofofze — APPROVED** [review]

File: reports.go [hunk 1]
```diff
@@ -34,3 +41,42 @@ func IndexPriceUpdatesBatchSubmitted(size int, tags ...Tags) {
 		s.Count("events.set_price_batch.size", int64(size), tagSpec, 1)
 	})
 }
+
+func Counter(metric string, value int64, tags ...Tags) {
```
Comment (hmoragrega) [inline reports.go:45]: we could have the generic version too

File: client.go [hunk 2]
```diff
@@ -173,44 +202,89 @@ func checkConfig(cfg *StatterConfig) *StatterConfig {
...
+	}
 }
 
 type mockStatter struct {
+	l         log.Logger
+	threshold time.Duration
 }
 
 func (s *mockStatter) Count(name string, value int64, tags []string, rate float64) error {
+	if s.threshold > 0 {
```
[the remark this answers is not in the document]
Comment (danidomi) [reply client.go:223]: I just copy & pasta from `indexer`!

from what I see if we go above the value of the threshold we do not log anymore for any metric

https://github.com/InjectiveLabs/injective-indexer/blob/dev/metrics/client.go#L156

File: reports.go (hunk 1 above)
Comment (danidomi) [reply reports.go:45]: done!

File: client.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (hmoragrega) [reply client.go:223]: but makes no sense for a counter, I would remove it

**Review by hmoragrega — APPROVED** [review]

File: client.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (danidomi) [reply client.go]: done ✅

**Pull request merged**